### PR TITLE
Added new functions to unregister readers, writers, and identifiers

### PR DESCRIPTION
**Pull request by astrofrog**
@taldcroft @MSeifert04 - I think you contributed the most edits to this file, so tagging you so you can review this. I'd like to get this in 2.0 if it's uncontroversial.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -56,6 +56,8 @@ New Features
 
 - ``astropy.io.registry``
 
+  - New functions to unregister readers, writers, and identifiers. [#6216]
```
Comment (pllim) [inline CHANGES.rst]: Off by one!

Do we need What's New for this?

Comment (astrofrog) [inline CHANGES.rst]: Oops: https://github.com/astropy/astropy-tools/commit/127fffc1104c9e0f3404ce1e9185ca440e9cf192 😊 

Don't think we need a what's new, I don't think this is really exciting enough to qualify.

File: astropy/io/registry.py [hunk 2]
```diff
@@ -324,6 +368,25 @@ def register_identifier(data_format, data_class, identifier, force=False):
...
+        The data format identifier.
+    data_class : classobj
+        The class of the object that can be read/written.
+    """
+
+    if (data_format, data_class) in _identifiers:
+        _identifiers.pop((data_format, data_class))
+    else:
+        raise IORegistryError("No identifier defined for format '{0}' and class '{1}'"
+                              ''.format(data_format, data_class.__name__))
```
Comment (eteq) [inline astropy/io/registry.py]: (nitpick) >80 chars

Comment (eteq) [inline astropy/io/registry.py]: I'll just fix this directly and merge.

**Pull request merged**